tabpy-server/server_tests/test_config.py

```diff
@@ -127,128 +152,97 @@ def setUp(self):
         self.addCleanup(os.chdir, self.cwd)
 
     def test_http(self):
-        self.fp.write(b'TABPY_TRANSFER_PROTOCOL = http')
+        self.fp.write("[TabPy]\n"
+                      "TABPY_TRANSFER_PROTOCOL = http".encode())
         self.fp.close()
 
-        settings, _ = get_config()
+        settings, _ = get_config(self.fp.name)
         self.assertEqual(settings['transfer_protocol'], 'http')
 
     def test_https_without_cert_and_key(self):
-        self.fp.write(b'TABPY_TRANSFER_PROTOCOL = https')
+        self.fp.write("[TabPy]\n"
```

Comment:
Pretty strange to call `encode` on the string here before writing. Much better options include just creating bytes instead of strings or potentially not opening the file in binary mode (latter might be more complicated, didn't look comprehensively)